code/test/general.js

```diff
@@ -0,0 +1,2306 @@
+const TestUtil = require('./util')
```

Comment:
Generally, break tests into small pieces and have each file focus on a single specialized area of features. This is too long and overlaps with many existing tests